src/parsers/array.ts

```diff
@@ -124,18 +145,34 @@ export const array: Parser<unknown[]> = {
         type: getTupleType(arrayType, fixedLength),
         buffer,
         value,
+        packed: fixedBytes.isType(arrayType) && tight,
+        tight,
+      });
+    }
+
+    if (packed) {
+      // For packed encoding, we don't need to encode the length of the array,
+      // so we can just encode the values.
+      return pack({
+        types: new Array(value.length).fill(arrayType),
+        values: value,
+        byteArray: buffer,
+        packed: fixedBytes.isType(arrayType) && tight,
```

Comment:
Not sure I understand the logic here 🤔

Reply:
This is a weird edge case in `ethereumjs-abi`. It disregards the required size (i.e., 32 bytes for arrays) and always encodes `bytesN` as their length.

https://github.com/ethereumjs/ethereumjs-abi/blob/ee3994657fa7a427238e6ba92a84d0b529bbcde0/lib/index.js#L479-L485

Reply:
I think we should add a comment that explains this!